### PR TITLE
fix low-bit onnx exporter

### PR DESCRIPTION
**Pull request by lapid92**

### CODE REVIEW

File: model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py [hunk 1]
```diff
@@ -70,6 +74,18 @@ def export(self) -> None:
 
         Logger.info(f"Exporting PyTorch fake quant onnx model: {self.save_model_path}")
 
+        # Replace float weight with wrapped quantized weights
+        for layer in self.model.modules():
+            if isinstance(layer, PytorchQuantizationWrapper):
+                weights_quantizers = list(layer.weights_quantizers.values())
+                if len(weights_quantizers) > 0:
+                    quantized_weight = torch.nn.Parameter(layer.get_quantized_weights()[KERNEL]).detach()
```
Comment (haihabi) [inline model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py:82]: What happens if we have more then Kernel-like biases...etc (hint you have the names inside the quantized wrapper).

File: model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py [hunk 2]
```diff
@@ -70,6 +74,18 @@ def export(self) -> None:
...
+                    delattr(layer, KERNEL)
+                    setattr(layer, KERNEL, torch.nn.Parameter(quantized_weight))
+                    linear_layer = getattr(layer, LAYER)
+                    delattr(linear_layer, KERNEL)
+                    setattr(linear_layer, KERNEL, torch.nn.Parameter(quantized_weight))
+                layer.weights_quantizers = []
```
Comment (haihabi) [inline model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py:88]: Should be a dict not a list

File: model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py [hunk 3]
```diff
@@ -70,6 +74,18 @@ def export(self) -> None:
...
+        for layer in self.model.modules():
+            if isinstance(layer, PytorchQuantizationWrapper):
+                weights_quantizers = list(layer.weights_quantizers.values())
+                if len(weights_quantizers) > 0:
+                    quantized_weight = torch.nn.Parameter(layer.get_quantized_weights()[KERNEL]).detach()
+                    delattr(layer, KERNEL)
+                    setattr(layer, KERNEL, torch.nn.Parameter(quantized_weight))
+                    linear_layer = getattr(layer, LAYER)
+                    delattr(linear_layer, KERNEL)
+                    setattr(linear_layer, KERNEL, torch.nn.Parameter(quantized_weight))
```
Comment (haihabi) [inline model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py:87]: Why do we need to override both? Does the first inference already do it?

File: model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py [hunk 4]
```diff
@@ -22,6 +22,9 @@
 from packaging import version
 
 # ONNX opset version 16 is supported from PyTorch 1.12
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_exporter/pytorch/fakely_quant_onnx_pytorch_exporter.py:24]: Move this comment to where it was (before the versioning check)

**Review by reuvenperetz — APPROVED** [review]

**Pull request merged**